frontend/wallet/src/render/components/SettingsPage.re

```diff
@@ -7,12 +9,74 @@ module Styles = {
       padding(`rem(2.)),
       backgroundColor(Theme.Colors.greyish(0.1)),
     ]);
+
+  let walletItemContainer =
+    style([
+      display(`flex),
+      flexDirection(`column),
+      backgroundColor(white),
+      padding(`rem(0.5)),
+      borderRadius(`px(6)),
+      border(`px(1), `solid, Theme.Colors.slateAlpha(0.5)),
+      width(`rem(28.)),
+    ]);
+
+  let walletItem =
+    merge([
+      Theme.Text.Body.regular,
+      style([
+        margin(`rem(0.5)),
+        color(Theme.Colors.midnight),
+        display(`flex),
+        alignItems(`center),
+      ]),
+    ]);
+
+  let walletName = style([width(`rem(12.5))]);
+
+  let walletChevron = style([color(Theme.Colors.teal)]);
+
+  let line =
+    style([
+      border(`zero, `none, transparent),
+      borderTop(`px(1), `solid, Theme.Colors.slateAlpha(0.5)),
+      width(`percent(100.)),
+    ]);
 };
 
-module SettingsQueryString = [%graphql {| query settings { version } |}];
+module SettingsQueryString = [%graphql
+  {| query getSettings {
+      version
+      ownedWallets {
+        publicKey @bsDecoder(fn: "Apollo.Decoders.publicKey")
+      }
+     }
+|}
+];
 
 module SettingsQuery = ReasonApollo.CreateQuery(SettingsQueryString);
 
+module WalletItem = {
+  [@react.component]
+  let make = (~publicKey) => {
+    let (addressBook, _) = React.useContext(AddressBookProvider.context);
+    let keyStr = PublicKey.toString(publicKey);
```

Comment:
nit: can you just use `key` since you pass it in already `key={PublicKey.toString(w##publicKey)}`

Reply:
The key field that's passed in is the react-specific key property for handling array diffing better, I'm not sure that it's great practice to use it and I don't think reason react would even let you

Reply:
makes sense, sounds good